File: 프로그래머스/lv0/120863. 다항식 더하기/다항식 더하기.py
def solution(polynomial):
    x, con = 0, 0
    
    # 공백을 기준으로 분리
    polynomial = polynomial.replace(' ', '').split('+')
    for i in polynomial:
        if 'x' in i:
            if len(i)>1:
                x+=int(i[:-1])
            else:
                x+=1
        else:
            con+=int(i)
    
    if x==0:
        if con==0:
            return '0'
        else:
            return str(con)
    elif x==1:
        if con==0:
            return 'x'
        else:
            return 'x + ' + str(con)
    else:
        if con==0:
            return str(x)+'x'
        else:
            return str(x)+'x + ' +str(con)

File: 프로그래머스/lv0/120863. 다항식 더하기/다항식 더하기.js
function solution(polynomial) {
    var a = 0, b = 0;
    var arr = polynomial.split(' + ');
    for (var k of arr){
        if (k.includes('x')){
            if (k.length===1){
                a+=1;
            }
            else{
                a+=Number(k.split('x')[0]);
            }
        }
        else{
            b+=Number(k);
        }
    }
    return (a===0)? ((b===0)?'0':""+b+""):((b===0)?((a===1)?'x':a+'x'):(a===1)?'x + '+b:a+'x + '+b);
}